### PR TITLE
feat: summarize insights in one place

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

We're summarizing insights in more than one place and (at least in the short term) it is getting more complicated

## Changes

Moves summarization into one place

part of #13927 

## How did you test this code?

👀

### CODE REVIEW

File: frontend/src/scenes/insights/InsightPageHeader.tsx [hunk 1]
```diff
@@ -106,19 +97,7 @@ export function InsightPageHeader({ insightId }: InsightSceneProps): JSX.Element
...
-                                      aggregationLabel,
-                                      cohortsById,
-                                      mathDefinitions
-                                  )
-                                : isFilterBasedInsight
-                                ? summarizeInsightFilters(filters, aggregationLabel, cohortsById, mathDefinitions)
-                                : // TODO placeholder for non insight viz queries
-                                  ''
-                        }
+                        placeholder={insightSummaryString({ insight, isUsingDataExploration })}
```
Comment (thmsobrmlr) [inline frontend/src/scenes/insights/InsightPageHeader.tsx:100]: Unfortunately this doesn't work as intended. `insight.filters` don't generally get updated by data exploration. Sometimes they do, but this is only due to happenstance when I opted to add a listener to an existing `setFilters` operation and updated the query from there. 

We could add general a `setFilters` action to the `setQuery` listener inside `insightDataLogic` to make that happen, but I was planning on going opposite direction and separating them further. I'm thinking we need some kind of hard cut, as otherwise nobody will know what's going on and what's updating what.

Basically values from `insightDataLogic/pathsDataLogic/...` should be used for data exploration and values from `insightLogic/trendsLogic/pathsLogic/...` should be used for non data exploration, as before.

Hope you understand what I wrote and happy to get on a call to discuss this.

Comment (pauldambra) [reply frontend/src/scenes/insights/InsightPageHeader.tsx:100]: Hmmm, it's definitely too coupled to what's in the other PR to make sense alone then 👍
